app/scripts/lib/marketing-email-client.js

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A client to talk to the basket marketing email server
+ */
+
+'use strict';
+
+define([
+  'lib/xhr'
+], function (xhr) {
+
+  function MarketingEmailClient(options) {
+    options = options || {};
+
+    var self = this;
+
+    self._xhr = options.xhr || xhr;
+    self._baseUrl = options.baseUrl;
+  }
+
+  MarketingEmailClient.prototype = {
+    _request: function (path, type, accessToken, data, headers) {
+      var url = this._baseUrl + path;
+
+      return this._xhr.oauthAjax({
+        url: url,
+        type: type,
+        accessToken: accessToken,
+        data: data,
+        headers: headers
+      })
+      .fail(function () {
+        // TODO - no server is listening just yet. Remove this when ready.
+        // ignore the error until the server is ready.
+        return {};
+      });
+    },
+
+    fetch: function (accessToken) {
+      return this._request('/lookup-user', 'get', accessToken);
+    },
+
+    optIn: function (accessToken, newsletterId) {
+      return this._request('/subscribe', 'post', accessToken, {
+        //jshint camelcase: false
+        newsletter_id: newsletterId
```

Comment:
For FxA users, their "lang" setting in basket is supposed to come from their FxA  account data, and is populated by a background process that transfers FxA account data over to basket.  There's some corner-cases here if we try to process the opt-in before the background transfer has completed.

I think the safest course of action will be to leave out "lang" on this request, and only enable the opt-in for en-US on initial release.  That way if there's a race, the worst that happens is they get default english newsletters until we set their actual language preference, which is what they wanted anyway.

We'll need to sanity-check this with @pmclanahan.